internal/pkg/deploy/cloudformation/stack/worker_service_integration_test.go

```diff
@@ -89,4 +92,24 @@ func TestWorkerService_Template(t *testing.T) {
 
 		require.Equal(t, string(wantedCFNParamsBytes), actualParams)
 	})
+
+	t.Run("invalid topic subscription should be rejected", func(t *testing.T) {
+		path := filepath.Join("testdata", "workloads", workerManifestPath)
+		manifestBytes, err := ioutil.ReadFile(path)
+		require.NoError(t, err)
+		mft, err := manifest.UnmarshalWorkload(manifestBytes)
+		require.NoError(t, err)
+		envMft, err := mft.ApplyEnv(subscriptionTestEnvName)
+		require.NoError(t, err)
+		v, ok := envMft.(*manifest.WorkerService)
+		require.True(t, ok)
+		badSubSerializer, err := stack.NewWorkerService(v, subscriptionTestEnvName, appName, stack.RuntimeConfig{
+			ServiceDiscoveryEndpoint: "test.my-app.local",
+			AccountID:                "123456789123",
+			Region:                   "us-west-2",
+		}, []string{"arn:aws:sns:us-west-2:123456789123:my-app-test-dogsvc-givesdogs", "arn:aws:sns:us-west-2:123456789123:my-app-test-dogsvc-giveshuskies"})
```

Comment:
how do these topics get passed here? why would these be different than what's provided in the manifest?

Reply:
These are generated by a call to the resource groups tagging API at `svc deploy` time. We unfortunately have to pass them all the way down into the stack package because that's the only place we can validate manifest inputs when we're not parsing flags.